eyeshade/controllers/publishers.js

```diff
@@ -243,6 +243,140 @@ v2.getBalance = {
 }
 
 /*
+   GET /v2/publishers/{publisher}/wallet
+       [ used by publishers ]
+ */
+
+v2.getWallet = {
+  handler: (runtime) => {
+    return async (request, reply) => {
+      const publisher = request.params.publisher
+      const currency = request.query.currency.toUpperCase()
+      const debug = braveHapi.debug(module, request)
+      const publishers = runtime.database.get('publishers', debug)
+      const settlements = runtime.database.get('settlements', debug)
+      const voting = runtime.database.get('voting', debug)
+      let amount, entries, entry, result, summary
+      let probi = new BigNumber(0)
+
+      summary = await voting.aggregate([
+        {
+          $match:
+          {
+            probi: { $gt: 0 },
+            publisher: { $eq: publisher },
+            altcurrency: { $eq: altcurrency },
+            exclude: false
+          }
+        },
+        {
+          $group:
+          {
+            _id: '$publisher',
+            probi: { $sum: '$probi' }
+          }
+        }
+      ])
+      if (summary.length > 0) probi = new BigNumber(summary[0].probi.toString())
+
+      summary = await settlements.aggregate([
+        {
+          $match:
+          {
+            probi: { $gt: 0 },
+            publisher: { $eq: publisher }
+          }
+        },
+        {
+          $group:
+          {
+            _id: '$publisher',
+            probi: { $sum: '$probi' }
+          }
+        }
+      ])
+      if (summary.length > 0) probi = probi.minus(new BigNumber(summary[0].probi.toString()))
+      if (probi.lessThan(0)) probi = 0
+
+      amount = runtime.currency.alt2fiat(altcurrency, probi, currency) || 0
+      result = {
+        rates: runtime.currency.rates[altcurrency],
+        contributions: {
+          amount: amount,
+          currency: currency,
+          altcurrency: altcurrency,
+          probi: probi.truncated().toString()
+        }
+      }
+
+      entries = await settlements.find({ publisher: publisher }, { sort: { timestamp: -1 }, limit: 1 })
+      console.log(JSON.stringify(entries, null, 2))
+      entry = entries && entries[0]
+      if (entry) {
+        probi = new BigNumber(entry.probi.toString())
+        result.lastSettlement = {
+          amount: runtime.currency.alt2fiat(altcurrency, probi, currency) || 0,
```

Comment:
should we also include the fiat value at the time of conversion in the settlement objects? maybe just for cards denominated in fiat?

Reply:
interesting idea. let me look into that.